pkg/catalog/catalog.go

```diff
@@ -157,7 +157,11 @@ func getRegistryDevfiles(registry Registry) (registryDevfiles []DevfileComponent
 
 		err = json.Unmarshal(jsonBytes, &devfileIndex)
 		if err != nil {
-			return nil, errors.Wrapf(err, "Unable to unmarshal the devfile index.json from %s", indexLink)
+			if err := util.CleanDefaultHTTPCacheDir(); err != nil {
+				log.Warning("Error while cleaning up cache dir.")
+			}
+			// should we be worried about infinite loop?
```

Comment:
Who ever reviews, please suggest if we should worry about infinite loop here?

Reply:
Thinking out loud - instead of odo automatically trying to clean the cache, would it make sense to have a command like `odo registry --clean-cache` to do this when user asks for it? Something along the lines of `dnf clean all`. We can then simply ask the user in the `errors.Wrapf` to execute that command when `odo create` fails due to this issue.

I am asking this based on the concern of infinite loop.

Reply:
Maybe try only once after cleaning. If it fails again, we can error out and ask the user to check the devfile, network or use `--clean-cache`.

Reply:
sure, we dont have a `--clean-cache` method right, that should be added?

Reply:
Agreed on try to clean only once. will do that change. Also my suggestion would be to add `odo registry clean-cache` instead of a flag. more consistent.
But also think about it, this invalidation is just applied to github registries which we might remove soon https://github.com/openshift/odo/issues/4666
So do we wanna go overboard on something that happens rarely and might be removed soon?

Reply:
> So do we wanna go overboard on something that happens rarely and might be removed soon?

OK let's skip the `--clean-cache` part for now.

Reply:
done